### PR TITLE
assets min value lovelace based

### PR DESCRIPTION
**Pull request by rinor**

### CODE REVIEW

**Review by bartfish — APPROVED** [review]
LGTM!

**Comment by rinor** [comment]
> Tested locally and works well 100

thank you

File: src/contract.ts [hunk 1]
```diff
@@ -103,29 +103,30 @@ export class AllowedListContract {
...
-  
-        id = isHexStrict(id) ? id.substring(2) : id; // if 0x is there, then remove it
+
+        // id = isHexStrict(id) ? id.substring(2) : id; // if 0x is there, then remove it
         assetsDetails.push({
-          id: id,
-          min: fromWei(details.minimumValue),
+          id: stripHexPrefix(id), // if 0x is there, then remove it
+          min: fromWei(details.minimumValue, "microether"), // gives back microether = lovelace (for main asset),
+          // TODO: this will affect also other tokens/assets, but is this correct for them?
```
Comment (SebastienGllmt) [inline src/contract.ts:123]: Is it okay to merge with this TODO?

File: src/contract.ts [hunk 2]
```diff
@@ -103,29 +103,30 @@ export class AllowedListContract {
...
     for (let id of assets) {
       try {
         const details = await this.bridgeContract.methods.tokenRegistry(id).call();
-  
+
         if (details instanceof Error) return details;
-  
-        id = isHexStrict(id) ? id.substring(2) : id; // if 0x is there, then remove it
+
+        // id = isHexStrict(id) ? id.substring(2) : id; // if 0x is there, then remove it
```
Comment (SebastienGllmt) [inline src/contract.ts:119]: We can delete this comment entirely

File: src/contract.ts (hunk 1 above)
Comment (NicolasDP) [reply src/contract.ts:123]: this is only going to work for `wAda` not for any other assets. This is why there is this comment.

Comment (NicolasDP) [reply src/contract.ts:123]: We need to check the details's data to check if this is `WMAIN` or something else. @bartfish can you do that please ?

Comment (bartfish) [reply src/contract.ts:123]: sure

**Pull request merged**